src/apphosting/githubConnections.ts

```diff
@@ -257,6 +259,32 @@ async function promptCloneUri(
   return { cloneUri, connection: cloneUriToConnection[cloneUri] };
 }
 
+/**
+ * Prompts the user for a GitHub branch and validates that the given branch
+ * actually exists. User is re-prompted until they enter a valid branch.
+ */
+export async function promptGitHubBranch(repoLink: devConnect.GitRepositoryLink) {
+  const { lookupMap } = await devConnect.listAllBranches(repoLink.name);
+
+  let branch = DEFAULT_GITHUB_BRANCH;
+  do {
+    branch = await promptOnce({
+      name: "branch",
+      type: "input",
+      default: DEFAULT_GITHUB_BRANCH,
```

Comment:
optional: personally I would inline this and leave file-level comments to values that are used in multiple places. IMO this is easier to read:

`promptOnce({default: "main"})` vs `promptOnce({default: DEFAULT_BRANCH})`